File: Three-fullstack-projects/eCommerce-springboot-angular/E-Commerce-Spring-Boot/src/main/java/com/rycoding/ecommerce/service/student/TeacherService.java
package com.rycoding.ecommerce.service.student;

import com.rycoding.ecommerce.dto.GeneralResponse;
import com.rycoding.ecommerce.dto.TeacherDTO;

import java.util.List;

public interface TeacherService {
    GeneralResponse addTeacher(Long userId, TeacherDTO teacherDto);

    List<TeacherDTO> getAllTeachers(Integer pageNo, Integer pageSize, String sortBy);

    void deleteTeacher(Long teacherId);

    GeneralResponse updateTeacher(Long teacherId, TeacherDTO teacherDto);

    TeacherDTO getTeacherById(Long teacherId);
}

File: Three-fullstack-projects/eCommerce-springboot-angular/Student-Managment-Angular/src/app/services/teacher-service/teacher.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { UserStorageService } from '../storage/user-storage.service';
import { Observable } from 'rxjs';

const BASIC_URL = "http://localhost:8080/";

@Injectable({
  providedIn: 'root'
})
export class TeacherService {

  constructor(private http: HttpClient) { }

  addTeacher(teacherDto: any) {
    return this.http.post(BASIC_URL + `api/teacher/${UserStorageService.getUserId()}`, teacherDto, {
      headers: this.createdAuthorizationHeader(),
    });
  }

  getAllTeachers(): Observable<any> {
    return this.http.get(BASIC_URL + `api/teachers`, {
      headers: this.createdAuthorizationHeader(),
    });
  }

  deleteTeacher(teacherId: any): Observable<any> {
    return this.http.delete(BASIC_URL + `api/teacher/${teacherId}`, {
      headers: this.createdAuthorizationHeader(),
    });
  }

  getTeacherById(teacherId: any): Observable<any> {
    return this.http.get(BASIC_URL + `api/getTeacherById/${teacherId}`, {
      headers: this.createdAuthorizationHeader(),
    });
  }

  updateTeacher(teacherId: any, teacherDto: any) {
    return this.http.put(BASIC_URL + `api/teacher/${teacherId}`, teacherDto, {
      headers: this.createdAuthorizationHeader(),
    });
  }

  createdAuthorizationHeader(): HttpHeaders {
    let authHeader: HttpHeaders = new HttpHeaders();
    return authHeader.set(
      "Authorization",
      "Bearer " + UserStorageService.getToken()
    )
  }


}
